### DISCUSSION TITLE
How to develop a custom plugin with javascript and publishing them to public folder?

### DISCUSSION THREAD

**Discussion by simonschaufi** [kimai/kimai#5244] [Questions & Answers] [answered]
I'm currently developing a custom plugin and I wonder how I can symlink the assets from my plugin to the public folder because the `assets:install` command copies them.

I did some research on symfony development myself already and couldn't find any solution.

**Comment by salvamaine** [comment]
Did you find anything? Right now I'm injecting JS using a Subscriber for ThemeEvent::JAVASCRIPT event and adding it using  $event->addContent($js); but it's not very convenient as the JS code must be included inside a string variable.

**Reply by simonschaufi** [reply]
No, I still haven't found a solution yet.

**Comment by kevinpapst** [comment]
What are you trying to achieve and what is the problem?

A plugin installer will copy the assets automatically: 
https://github.com/kimai/kimai/blob/main/src/Command/AbstractBundleInstallerCommand.php#L45

If you want to inject global javascript, then these `ThemeEvent` is indeed the correct place.
If you use your own views, you simply overwrite the `{% block javascript %}`.

**Reply by simonschaufi** [reply]
The "copy" is my problem because I want to work on my js and see instant changes on the app in the brower without coping the file all the time. I wonder how you develop and work with js? It would be great to have some documentation about it!

**Reply by kevinpapst** [reply, marked as answer]
For development you can try to use `bin/console assets:install --symlink`

**Reply by salvamaine** [reply]
If we write a plugin installer, is it executed automatically just being an extension of AbstractBundleCommand or is there more configuration involved? And in the case of a plugin upgrade is it executed automatically or the copy must be executed manually with asset:install ?

**Reply by kevinpapst** [reply]
> If we write a plugin installer, is it executed automatically just being an extension of AbstractBundleCommand or is there more configuration involved? 

Each bundle installer needs to be executed manually. That will change with the next major version, but for now, you have to call `bin/console kimai:bundle:foo:install`

> And in the case of a plugin upgrade is it executed automatically or the copy must be executed manually with asset:install ?

Assets are automatically installed when running the installer.
